pkg/cloudprovider/provider/openstack/provider.go

```diff
@@ -177,6 +178,31 @@ func (p *provider) getConfigAuth(c *Config, rawConfig *openstacktypes.RawConfig)
 	return nil
 }
 
+func (p *provider) resolveNetworks(cfg *Config) ([]string, error) {
+	var networks []string
+
+	// Validation: ensure both network and networks are not set simultaneously
+	if cfg.Network != "" && len(cfg.Networks) > 0 {
+		return nil, fmt.Errorf("cannot specify both 'network' and 'networks' fields simultaneously, use only one")
```

Comment:
But why not? Technically we can easily prepend `cfg.Network` to `cfg.Networks`, and later in the code use `cfg.Networks` ONLY.